### PR TITLE
Added disabler SMG!

### PR DESCRIPTION
**Pull request by Stanislav4ix**
## About the PR
Added disabler SMG, you can print it on security techfab when you researched advanced riot control technologies. 

## Why / Balance
Will be useful for security personnel.

## Technical details
Disabler SMG has fire rate 4 shoots per second. Disabler SMG uses new type of ammo - that has 2 heat damage and 15 stamina damage. That means that you should fire a shot 7 times for make stamina crit to someone. It has charge for 30 shots and full/semi-auto firing modes.

## Media
image
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl: Stanislav4ix
Added disabler SMG! You should research advanced riot control and then you will be able to print it on security techfab.

### CODE REVIEW

**Comment by K-Dynamic** [comment]
More sec content is good, actually

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml [hunk 1]
```diff
@@ -596,3 +596,39 @@
   - type: Appearance
   - type: StaticPrice
     price: 750
+
```
Comment (Errant-4) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml:599]: would be generally more readable if you added this entire block right after/near the disabler, instead the end of the file

File: Resources/Prototypes/Entities/Structures/Machines/lathe.yml [hunk 2]
```diff
@@ -612,6 +612,8 @@
       - Truncheon
       - VoiceTrigger
       - WeaponAdvancedLaser
+      - WeaponDisablerPractice
```
Comment (Errant-4) [inline Resources/Prototypes/Entities/Structures/Machines/lathe.yml:615]: Why is WeaponDisablerPractice duplicated here? Its already under static recipes and there is no mention of this on the PR

File: Resources/Prototypes/Recipes/Lathes/security.yml [hunk 3]
```diff
@@ -463,8 +463,7 @@
     Steel: 500
     Glass: 100
     Plastic: 200
-
-- type: latheRecipe
+
```
Comment (Errant-4) [inline Resources/Prototypes/Recipes/Lathes/security.yml:466]: You broke the uranium shell lathe recipe by accidentally deleting it's first line

**Review by Errant-4 — CHANGES_REQUESTED** [review]
I have some suggestions re the code as is

File: Resources/Prototypes/Recipes/Lathes/security.yml [hunk 4]
```diff
@@ -464,7 +464,7 @@
     Glass: 100
     Plastic: 200
 
-- type: latheRecipe
+- type: latheRecipe
```
Comment (TK-A369) [inline Resources/Prototypes/Recipes/Lathes/security.yml:467]: I don't think those spaces at line end were intended.

**Comment by brainfood1183** [comment]
I think the smg could be a possibility if it fires much faster but each shot is also 1/3 the power of a normal disabler. Using fire rate and disabler shots on screen to disable someone if you aint that accurate at the cost of requiring more hits.

**Comment by asperger-sind** [comment]
This needs a video because it's not really clear how fast this fires and downs people and there is no mention of how the spread is handled.

Imo this is fine, if silly and allows for really low TT*K*s on armored targets, just needs a bit of spread so it's not overtuned.

**Comment by Stanislav4ix** [comment]
> This needs a video because it's not really clear how fast this fires and downs people and there is no mention of how the spread is handled.
> 
> Imo this is fine, if silly and allows for really low TT_K_s on armored targets, just needs a bit of spread so it's not overtuned.

I will make a video, but for now I will say that it takes the same amount of time (not shots) as a disabler for a person to fall

**Comment by Emisse** [comment]
this is sick but im wondering if this should go into t3 tech or not, might need discussion

**Comment by keronshb** [comment]
Adding a rapid fire, mass producible, disabler is a huge balancing issue. While it could be potentially balanced in the current PR, it could become completely overpowered with one or two tweaks.

**Comment by HoidC** [comment]
> Adding a rapid fire, mass producible, disabler is a huge balancing issue. While it could be potentially balanced in the current PR, it could become completely overpowered with one or two tweaks.

can I ask what you mean by this? any gun can indeed become overpowered in a few tweaks, such is the nature of a weapon of that variety

File: Resources/Prototypes/Entities/Structures/Machines/lathe.yml (hunk 2 above)
Comment (Errant-4) [reply Resources/Prototypes/Entities/Structures/Machines/lathe.yml:614]: This duplication of WeaponDisablerPractice is still here. Is this intentional?

**Comment by Stanislav4ix** [comment]
> Adding a rapid fire, mass producible, disabler is a huge balancing issue. While it could be potentially balanced in the current PR, it could become completely overpowered with one or two tweaks.

It is perfectly balanced, because it takes the same amount of time for a person to fall, and during this time it will cause less damage than the disabler. One disabler shot deals 5 heat damage and a disabler SMG shot deals 2 heat damage. That is, 4 shots from the disabler (that's enough for a person to fall) will cause 20 thermal damage, while 7 shots from the disabler SMG will cause only 14 thermal damage. Therefore, I believe that this weapon is perfectly balanced in two ways

**Comment by keronshb** [comment]
> > Adding a rapid fire, mass producible, disabler is a huge balancing issue. While it could be potentially balanced in the current PR, it could become completely overpowered with one or two tweaks.
> 
> can I ask what you mean by this? any gun can indeed become overpowered in a few tweaks, such is the nature of a weapon of that variety

> It is perfectly balanced, because it takes the same amount of time for a person to fall, and during this time it will cause less damage than the disabler. One disabler shot deals 5 heat damage and a disabler SMG shot deals 2 heat damage. That is, 4 shots from the disabler (that's enough for a person to fall) will cause 20 thermal damage, while 7 shots from the disabler SMG will cause only 14 thermal damage. Therefore, I believe that this weapon is perfectly balanced in two ways

Specifically with guns that can stun, it can be a huge problem to have something so widely available. An old fun combo is a toy LMG + foam riot darts. Both mass producible and they can stun relatively fast. Thankfully the way to mass produce them was locked behind a lot of tech trees and were stuck to lathes that required plastic and you needing to manually reload the mag.

Putting this kind of weapon in the sec lathe is still a huge balance issue.

While any gun can be tweaked yes, it adds another one to the pool. Having too many to keep track of, at least player facing ones, can be a huge pain to balance for future considerations. Let's say if stamina damage was tweaked or disabler blasts themselves were tweaked, it could make this extremely broken.

I do not believe we need any more stun weapons and should focus on what we have currently.

If we absolutely need this, as much as I hate cargo reliance, it should be locked behind ordering in cargo with a few crates. At least then it could be controlled via ordering instead of allowing a mass print.

**Comment by chromiumboy** [comment]
Adding a second ranged stunning weapon seems hard to justify, especially one that's mass producible

What role is this weapon meant to serve and why can't the disabler be used to fulfil that role? Simply saying "it'll be useful" isn't sufficient justification IMO

**Comment by Emisse** [comment]
I could see a compromise being to give this to the armory, it has same time to stun as disabler but is more forgiving since it rapid fires

**Comment by HoidC** [comment]
> I could see a compromise being to give this to the armory, it has same time to stun as disabler but is more forgiving since it rapid fires

disablers are balanced right now by not being weapons you can guarantee landing a bunch of shots with - they're  better for a chase, but aren't impossible to avoid if you juke well

this weapon circumvents that by being an automatic weapon, which is something you can't really juke unless you're practically offscreen

this being a roundstart weapon is pretty imbalanced

**Comment by Emisse** [comment]
if theres only one in the armory it would be fine imo, considering sec is generally underpowered right now

**Comment by K-Dynamic** [comment]
Just do a test merge and revert if it's too busted

Personally a disabler SMG would be more forgiving to use with high ping compared to the normal disabler

**Comment by mirrorcult** [comment]
im kind of interested in seeing this, definitely as a tech i think though

**Comment by Aisu9** [comment]
> > I could see a compromise being to give this to the armory, it has same time to stun as disabler but is more forgiving since it rapid fires
> 
> disablers are balanced right now by not being weapons you can guarantee landing a bunch of shots with - they're better for a chase, but aren't impossible to avoid if you juke well
> 
> this weapon circumvents that by being an automatic weapon, which is something you can't really juke unless you're practically offscreen
> 
> this being a roundstart weapon is pretty imbalanced

isn't that a whole point of.... research and equipment upgrade? disabler smg aren't round start and are locked behind research, they are rapid fire and are easier to use for people with poor aim. the stun time is still the same so it's only an improvement regarding accuracy forgiveness

**Comment by deltanedas** [comment]
this should have lower ttk and be t3 research imo

**Comment by Aisu9** [comment]
i guess you get to stun 2 additional people before needing to recharge it
disabler : 4 shots to stun - 10 shots in battery = 2 stun
disabler smg : 7 shots to stun - 30 shots in battery = 4 stuns
and both require 2 seconds of constant firing in order to stun, and all shots need to lands(i suppose the smg have some deviation)

**Comment by Stanislav4ix** [comment]
> this should have lower ttk and be t3 research imo

makes no sense, because in t3 we have taser

**Comment by Emisse** [comment]
that was removed

**Comment by Aisu9** [comment]
the only god damn difference between a disabler and the smg disabler is the accuracy forgiveness AND more stun critable target before recharge, this does not need to be locked behind T3, because all T3 are meant to be balance breaking in some ways

**Comment by Stanislav4ix** [comment]
> that was removed

ok then i will move disabler SMG to t3 research

**Comment by Flareguy** [comment]
If this functions like it does in /tg/ (a sidegrade to the regular disabler) then it should not be a T3 at all. T3 techs are meant for overpowered and round-changing techs, which the automatic disabler most *certainly* does not fit the criteria of.

**Comment by Aisu9** [comment]
> > that was removed
> 
> ok then i will move disabler SMG to t3 research

the SMG isn't worth T3 placement at all, T3 research are meant for items that doesn't require balancing, as of right now it's just a normal disabler that is more forgiving for the user accuracy

**Comment by Stanislav4ix** [comment]
> > > that was removed
> > 
> > ok then i will move disabler SMG to t3 research
> 
> the SMG isn't worth T3 placement at all, T3 research are meant for items that doesn't require balancing, as of right now it's just a normal disabler that is more forgiving for the user accuracy

yes, and no one will want to research arsenal technology just for disabler SMG

**Comment by deltanedas** [comment]
it should be made worth using then, just a disabler for people with krill issue (pro hos that cant aim) wouldnt be worth getting.

**Comment by Aisu9** [comment]
> > > > that was removed
> > > 
> > > 
> > > ok then i will move disabler SMG to t3 research
> > 
> > 
> > the SMG isn't worth T3 placement at all, T3 research are meant for items that doesn't require balancing, as of right now it's just a normal disabler that is more forgiving for the user accuracy
> 
> yes, and no one will want to research arsenal technology just for disabler SMG

because you think science will research a T3 and lock themselves out of every other T3 tech branches? T2 is fine and doesn't lock science. imagine there could be sec officers with that bad boy AND bag of holding

**Comment by deltanedas** [comment]
typically a thing like this would only be researched during nukies or revs. outside of that the mentality is sec already has disablers so they dont need anything else, up until nukies knock on our door then research every arsenal tech.

**Comment by Aisu9** [comment]
> it should be made worth using then, just a disabler for people with krill issue (pro hos that cant aim) wouldnt be worth getting.

every "upgrades" doesn't need to be +1 in every stats you know?

**Comment by HoidC** [comment]
> > > that was removed
> > 
> > 
> > ok then i will move disabler SMG to t3 research
> 
> the SMG isn't worth T3 placement at all, T3 research are meant for items that doesn't require balancing, as of right now it's just a normal disabler that is more forgiving for the user accuracy

a normal disabler that does not require good aim is a very very very strong weapon as that is the downside of a disabler

**Comment by Aisu9** [comment]
> > > > that was removed
> > > 
> > > 
> > > ok then i will move disabler SMG to t3 research
> > 
> > 
> > the SMG isn't worth T3 placement at all, T3 research are meant for items that doesn't require balancing, as of right now it's just a normal disabler that is more forgiving for the user accuracy
> 
> a normal disabler that does not require good aim is a very very very strong weapon as that is the downside of a disabler

but both weapons needs 2 seconds of constant firing to stun, the fire rate only balance the lower stun damage you doofus

**Comment by Aisu9** [comment]
and the deviation wont guaranty you'll land all your shots despite the fire rate

**Comment by HoidC** [comment]
> > > > > that was removed
> > > > 
> > > > 
> > > > ok then i will move disabler SMG to t3 research
> > > 
> > > 
> > > the SMG isn't worth T3 placement at all, T3 research are meant for items that doesn't require balancing, as of right now it's just a normal disabler that is more forgiving for the user accuracy
> > 
> > 
> > a normal disabler that does not require good aim is a very very very strong weapon as that is the downside of a disabler
> 
> but both weapons needs 2 seconds of constant firing to stun, the fire rate only balance the lower stun damage you doofus

landing four shots in a row with a disabler is rather difficult on a strafing and fleeing suspect because of the nature of this game’s combat 

landing seven shots with an automatic weapon is an easier task

you are probably going to down them quicker because it is more forgiving to miss shots and it is easier to land shots with an automatic weapon

**Comment by HoidC** [comment]
this is a pointless argument I am banging my head against a brick wall and the relevant people have given this a go ahead so I will not rely further have a good day

**Comment by Stanislav4ix** [comment]
> > > > > > that was removed
> > > > > 
> > > > > 
> > > > > ok then i will move disabler SMG to t3 research
> > > > 
> > > > 
> > > > the SMG isn't worth T3 placement at all, T3 research are meant for items that doesn't require balancing, as of right now it's just a normal disabler that is more forgiving for the user accuracy
> > > 
> > > 
> > > a normal disabler that does not require good aim is a very very very strong weapon as that is the downside of a disabler
> > 
> > but both weapons needs 2 seconds of constant firing to stun, the fire rate only balance the lower stun damage you doofus
> 
> landing four shots in a row with a disabler is rather difficult on a strafing and fleeing suspect because of the nature of this game’s combat 
> 
> landing seven shots with an automatic weapon is an easier task
> 
> you are probably going to down them quicker because it is more forgiving to miss shots and it is easier to land shots with an automatic weapon
> 
> 

dude, it's SMG.  If you think like that, the full-auto shooting mode for the Bulldog makes it very easy to kill someone.  Like, this is SMG, what else should we expect from it?  And you can also switch between semi-auto and full-auto shooting modes

**Comment by Aisu9** [comment]
plus keep in mind that disablers are: roundstart, purchasable in bulk from cargo, unlockable in secfab to print them using resources collected or requested by cargo.
making the SMG T2 is honestly stretching the upgrade format just a little bit, but overall fells like it should still require some work

**Comment by IProduceWidgets** [comment]
I've toyed around with rapid fire disablers as admemes  few times. My 2c, is that they're not really that much amazinger.

That said, what I think would be better, is changing the burst fire rate to actually fire a quick burst but then have a longer cooldown between bursts. Its the ideal case for this sorta thing. 

Unfortunately right now burst fire is just full auto if you click fast, or terrible. It needs to be reworked to have a real burst functionality.

**Comment by asperger-sind** [comment]
> ...considering sec is generally underpowered right now - emisse

Sec isn't underpowered, they've lost their 2-hit batons and better armor but gained roundstart pistols, 12 damage knives, and things like secfab, while retaining the overpowered equipment-check mess like flashes and flashbang, playing sec also doesn't feel like the dept has weak gear for dealing with tasks they are assigned.

Having the disabler SMG (will be referred to as "SMG" from now on) roundstart invalidates its purpose as a flat upgrade kind of item, it is also unclear what is the use case for the SMG on roundstart, which may just lead to the HoS snagging it roundstart.

Now unrelated to quote, the SMG doesn't exactly have the same TTK as the normal disabler in practice, in theory it does as a disabler shoots out 30 stamina damage twice a second, while the SMG shoots out 15 four times a second, when you start accounting for player input this becomes more evident, the most obvious example being how disablers aren't full auto and how people miss disabler shots very often, at this point the SMG is looking more like an upgrade because you actually get to utilize the really low TTK instead of hitting 2 shots out of 5 with ever so slight delays.

This item remains as a sidegradey upgrade to the disabler, it's better in most use cases and a lot more forgiving, having it as T1 raises potential issues with how these can be mass produced, having it as T3 has it underpowered and it'll never see the sun because nobody will research it, so the compromise would be having it as T2, preferably placed in the same research as some other sec items because the item is middling between T1 and T2 in terms of strength. I agree with this compromise.

**Comment by Emisse** [comment]
too many words not enough merg...

**Pull request merged**